samples/net/nats/src/main.c

```diff
@@ -202,10 +203,13 @@ static void write_led(const struct nats *nats,
 	}
 
 	pubstate = state ? "on" : "off";
-	nats_publish(nats, "led0", 0, msg->reply_to, 0,
-		     pubstate, strlen(pubstate));
-
-	printk("*** Turning LED %s\n", pubstate);
+	ret = nats_publish(nats, "led0", 0, msg->reply_to, 0,
+			   pubstate, strlen(pubstate));
+	if (ret < 0) {
+		NET_DBG("Failed to publish: %d", ret);
```

Comment:
Please don't use NET_DBG here - it's not a debugging message, it's real error. I'd recommend to just use printk(), like used below.